File: app/src/main/java/com/xando/chefsclub/Search/Core/ItemsClickListenerGetter.java
package com.xando.chefsclub.Search.Core;

import android.support.annotation.Nullable;

import com.mikepenz.fastadapter.items.AbstractItem;
import com.mikepenz.fastadapter.listeners.ClickEventHook;
import com.mikepenz.fastadapter.listeners.OnClickListener;

public interface ItemsClickListenerGetter<Item extends AbstractItem> {

    @Nullable
    ClickEventHook<Item> getClickEventHookInstance();

    @Nullable
    OnClickListener<Item> getClickItemListenerInstance();
}

File: app/src/main/java/com/xando/chefsclub/Search/Core/ItemsClickListenerGetter.kt
package com.xando.chefsclub.Search.Core

import android.support.v7.widget.RecyclerView
import com.mikepenz.fastadapter.IItem
import com.mikepenz.fastadapter.listeners.ClickEventHook
import com.mikepenz.fastadapter.listeners.OnClickListener

interface ItemsClickListenerGetter<Item : IItem<out Any, out RecyclerView.ViewHolder>> {

    val clickEventHook: ClickEventHook<Item>?
    val clickItemListener: OnClickListener<Item>
}